integration/utils/contLoadGenerator.go

```diff
@@ -304,76 +263,47 @@ func (lg *ContLoadGenerator) createAccounts(num int) error {
+	var i int
+	ch, err := lg.sendTx(-1, createAccountTx)
 	if err != nil {
 		return err
 	}
+	<-ch
 
-	executed := make(chan struct{})
-
-	var i int
 	log := lg.log.With().Str("tx_id", createAccountTx.ID().String()).Logger()
-	wait := lg.txTracker.AddTx(
-		createAccountTx.ID(),
-		txCallbacks{
-			onExecuted: func(txID flowsdk.Identifier, res *flowsdk.TransactionResult) {
-				log.Trace().
-					Str("status", res.Status.String()).
-					Msg("account creation tx executed")
-
-				if res.Error != nil {
-					log.Error().
-						Err(res.Error).
-						Msg("account creation tx failed")
-				}
-
-				for _, event := range res.Events {
-					log.Trace().
-						Str("event_type", event.Type).
-						Str("event", event.String()).
-						Msg("account creation tx event")
-
-					if event.Type == flowsdk.EventAccountCreated {
-						accountCreatedEvent := flowsdk.AccountCreatedEvent(event)
-						accountAddress := accountCreatedEvent.Address()
-
-						log.Trace().
-							Hex("address", accountAddress.Bytes()).
-							Msg("new account created")
-
-						signer, err := crypto.NewInMemorySigner(privKey, accountKey.HashAlgo)
-						if err != nil {
-							panic(err)
-						}
-
-						newAcc := newFlowAccount(i, &accountAddress, accountKey, signer)
-						i++
-
-						lg.accounts = append(lg.accounts, newAcc)
-						lg.availableAccounts <- newAcc
-
-						log.Trace().
-							Hex("address", accountAddress.Bytes()).
-							Msg("new account added")
-					}
-				}
-				close(executed)
-			},
-			onExpired: func(_ flowsdk.Identifier) {
-				log.Error().Msg("setup transaction (account creation) has expired")
-			},
-			onTimeout: func(_ flowsdk.Identifier) {
-				log.Error().Msg("setup transaction (account creation) has timed out")
-			},
-		},
-		120*time.Second,
-	)
+	result, err := lg.flowClient.GetTransactionResult(context.Background(), createAccountTx.ID())
+	if err != nil {
+		return fmt.Errorf("failed to get transactions result: %w", err)
+	}
 
-	select {
-	case <-wait:
-	case <-executed:
+	log.Trace().Str("status", result.Status.String()).Msg("account creation tx executed")
+	if result.Error != nil {
+		log.Error().Err(result.Error).Msg("account creation tx failed")
 	}
 
+	for _, event := range result.Events {
+		log.Trace().Str("event_type", event.Type).Str("event", event.String()).Msg("account creation tx event")
+
+		if event.Type == flowsdk.EventAccountCreated {
+			accountCreatedEvent := flowsdk.AccountCreatedEvent(event)
+			accountAddress := accountCreatedEvent.Address()
+
+			log.Trace().Hex("address", accountAddress.Bytes()).Msg("new account created")
+
+			signer, err := crypto.NewInMemorySigner(privKey, accountKey.HashAlgo)
+			if err != nil {
+				panic(err)
+			}
+
+			newAcc := newFlowAccount(i, &accountAddress, accountKey, signer)
+			i++
+
+			lg.accounts = append(lg.accounts, newAcc)
+			lg.availableAccounts <- newAcc
+
+			log.Trace().Hex("address", accountAddress.Bytes()).Msg("new account added")
```

Comment:
return nil here (can the result contain multiple eventAccountCreated events)?

Reply:
yep, i think it will have exactly `i` accounts so returning here will break stuff.

Reply:
in that case, we should verify num accounts actually created = num asked?